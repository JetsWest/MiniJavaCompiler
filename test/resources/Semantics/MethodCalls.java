class SomeClass {
    public static void main(String[] args) {
        System.out.println(0);
    }
}

class A {
    public int foo() {
        return 0;
    }

    public int bar(int x) {
        return 0;
    }

    public int baz(int x, boolean y) {
        return 0;
    }

    public int bop() {
        int i;

        i = this.foo();
        i = this.bar(1);
        i = this.baz(1, true);

        i = this.foo(1);
        i = this.bar();
        i = this.bar(true);
        i = this.baz(1, 2);
        i = this.baz(1, true, 1);

        return 0;
    }
}