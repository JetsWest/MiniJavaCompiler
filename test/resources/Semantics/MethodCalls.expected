(Line 27) Incorrect number of parameters.
(Line 28) Incorrect number of parameters.
(Line 29) Incorrect parameter type at position 1.
(Line 30) Incorrect parameter type at position 2.
(Line 31) Incorrect number of parameters.